src/changelog/ticketChangelog.ts

```diff
@@ -40,36 +39,48 @@ export const getProjectScope = (
 export const getTagsScope = (ticket: GenericTicket): E.Either<Error, Scope> => {
```

Comment:
I suggest to refactor following this shape (in order to avoid nesting and lifting)

```ts
const validateLabel = O.of; // you custom validation logic

const getValidLabels = flow(
  (t: GenericTicket) => t.tags,
  RA.map(validateLabel),
  RA.compact
);

export const getTagsScope: (ticket: GenericTicket) => E.Either<Error, Scope> =
  flow(
    getValidLabels, // but I think that we can do a straighter validation
    E.of,
    E.filterOrElse(
      (scopes) => scopes.length > 1,
      () => new Error("Multiple labels match the expression blablabla.")
    ),
    E.chainOptionK(
      () => new Error("No labels match the expression blablabla.")
    )(RA.head),
    E.chainOptionK(
      () => new Error("The scope of the tag is not present in the allowed list")
    )((label) => pipe(tagToScope, R.lookup(label)))
  );
  
```